### PR TITLE
wip: use exactly_n and exactly_one from salt.utils.data

### PR DESCRIPTION
**Pull request by major0**
### What does this PR do?

Code maintenance replacing old usage of `botomod._exactly_one` with `salt.utils.data.exactly_one`

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- N/A Docs
- [X] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- N/A Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by major0** [comment]
@Akm0d Is there some sort of guidance for how to write a test case for a deleted interface?

**Comment by sagetherage** [comment]
@waynew maybe we can add this to the Test Clinic for this week or next?

**Comment by Akm0d** [comment]
Yes @waynew is an excellent tutor for writing tests and streams about it on twitch regularly

File: changelog/58708.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+**fixed:**
+
+Code maintenance replacing usage of `botomod._exactly_one` with `salt.utils.data.exactly_one`.
```
Comment (waynew) [inline changelog/58708.md:3]:
```suggestion
Replaced usage of `botomod._exactly_one` with `salt.utils.data.exactly_one`.
```

**Review by waynew — CHANGES_REQUESTED** [review]
The filename should also be `58708.changed`

As discussed on twitch - not sure if we need to deprecate the removed/now unused functions, but either approach should have another 58708.removed or .deprecated with the functions that were either deprecated or removed.

File: changelog/58708.md (hunk 1 above)
Comment (major0) [reply changelog/58708.md:3]: Submitted a PR to deprecate these instead of removing them: https://github.com/saltstack/salt/pull/58741

Comment (waynew) [reply changelog/58708.md:3]: @major0 should we go ahead and close this PR then?

**Comment by waynew** [comment]
@major0 the last note that I have to myself on this is that it should be removed... if that's not correct, let me know :joy: I'm just going to go ahead and close this for now.

**Pull request closed without merging**